presto-hive/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -1987,8 +1994,28 @@ public String getDescription()
 
         public void run(ExtendedHiveMetastore metastore)
         {
-            metastore.createTable(table, privileges);
-            done = true;
+            try {
+                metastore.createTable(table, privileges);
+                done = true;
+            }
+            catch (Throwable t) {
+                try {
+                    Optional<Table> remoteTable = metastore.getTable(table.getDatabaseName(), table.getTableName());
+                    // getPrestoQueryId(partition) is guaranteed to be non-empty. It is asserted in the constructor.
+                    if (remoteTable.isPresent() && getPrestoQueryId(remoteTable.get()).equals(getPrestoQueryId(table))) {
+                        done = true;
+                    }
+                }
+                catch (Throwable ignored) {
```

Comment:
Similar to above, I don't think this should ignore `Throwable`, since it could be `Error`. This should be `PrestoException` or `RuntimeException`.